### PR TITLE
Maya/per 10055 groups api

### PR DESCRIPTION
**Pull request by maya-barak**

### CODE REVIEW

File: docs/api/rebac/groups.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+sidebar_position: 7
+title: Groups API
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Groups API lets you manage your objects in a group. You can create, update, delete, and list groups.
```
Comment (GonPessach) [inline docs/api/rebac/groups.mdx:9]: Groups API allows you to **add** and manage your objects in a group....

File: docs/api/rebac/groups.mdx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Each group will have roles associated with it, and every user in the group will inherit the roles of the group.
+The group will get these roles from the resource that it is associated with.
```
Comment (GonPessach) [inline docs/api/rebac/groups.mdx:11]: not sure what is the intention in this sentence

File: docs/api/rebac/groups.mdx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+sidebar_position: 7
+title: Groups API
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Groups API lets you manage your objects in a group. You can create, update, delete, and list groups.
+Each group will have roles associated with it, and every user in the group will inherit the roles of the group.
```
Comment (GonPessach) [inline docs/api/rebac/groups.mdx:10]: maybe something on the lines of "You can use groups to organize users and assign roles collectively by assigning roles to the group. Each group can have roles associated with it, and users in the group can inherit these roles from the resource it is linked to."

File: docs/api/rebac/groups.mdx (hunk 1 above)
Comment (GonPessach) [reply docs/api/rebac/groups.mdx:9]: here maybe I would mention what those objects could be: 
Users, resources, other groups etc.

File: docs/api/rebac/groups.mdx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
+sidebar_position: 7
+title: Groups API
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+The Groups API allows you to manage your objects within a group. You can create, update, delete, and list groups.
```
Comment (orweis) [inline docs/api/rebac/groups.mdx:9]:
```suggestion
The Groups API allows you to manage your resources and users within groups. You can create, update, delete, and list groups.
```

File: docs/api/rebac/groups.mdx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+Groups enable you to gather resource, resource instances and also allow you to manage ReBAC permissions for a group of users, and have associated roles.
```
Comment (orweis) [inline docs/api/rebac/groups.mdx:10]:
```suggestion
Groups enable you to easily group together resources, resource-instances, and users. In addition groups simplify ReBAC permissions for users, by allowing roles to be assigned to the groups themselves.
```

File: docs/api/rebac/groups.mdx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+You can assign roles to a group and then add users to the group through the Groups API.
+The roles assigned to the group will derive to the users within the group.
```
Comment (orweis) [inline docs/api/rebac/groups.mdx:12]:
```suggestion
The users within the group will derive roles form the roles assigned to the group .
```

File: docs/api/rebac/groups.mdx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+
+## Example Usage for User Groups
+
+We have a group called 'company_1', that will be represented with resource 'company_1', resource instance 'employees', and role 'member'.
+Adittionly, we will have another resource 'marketing' with resource instance 'social_media' and role 'editor'.
+Once we set the role 'editor' for the company_1 group, this is the process that will happen in the API:
+- create role derivation from 'company_1#member' to 'marketing#editor'
+- create a relation between 'company_1' and 'marketing' with a relation between 'group_company_1'
+- create relationship between 'company_1#employees' and 'marketing#social_media'.
```
Comment (orweis) [inline docs/api/rebac/groups.mdx:24]:
```suggestion
Let's say we want to represent a company and its employees as a group, and then assign all the employees access to a shared marketing resource (social-media) via the group.  We'll have a group type named 'company' (Which is represented as a resource named 'company'), and an instance of a specific company 'Acme'.  Users are assigned to `Acme` via a role called `member`
Additionally, we will have a resource-instance of type 'marketing' - 'social_media' which supports the role 'editor'.
Once we assign the role 'editor' to  the `Acme` group, the following will by automated by API:
- create role derivation from 'company#member' to 'marketing#editor'
- create a relation between 'company:Acme' and 'marketing:social-media' with a relationship type 'group'
```

File: docs/api/rebac/groups.mdx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  --data-raw '{"resource": "marketing", "resource_instance": "social_media", "role": "editor", tenant:"business"}' \
+```
+
+To add users to a group, make a PUT request to the following with the required data:
+
+```bash
+curl 'https://api.permit.io/v2/facts/{project_id}/{env_id}/groups/{user_id}' \
+    -H 'authorization
+      --data-raw '{"tenant":"business"}' \
+    ```
```
Comment (orweis) [inline docs/api/rebac/groups.mdx:54]: These will also need to be adjusted accordingly

**Review by orweis — CHANGES_REQUESTED** [review]
I'm confused here... Let's discuss

**Pull request merged**